localstack/services/cloudformation/service_models.py

```diff
@@ -1034,6 +1034,31 @@ def _create(resource_id, resources, resource_type, func, stack_name, *args, **kw
         return {"create": {"function": _create}}
 
 
+class IAMUser(GenericBaseModel):
+    @staticmethod
+    def cloudformation_type():
+        return "AWS::IAM::User"
+
+    def get_physical_resource_id(self, attribute=None, **kwargs):
+        return self.props.get("UserName")
+
+    def get_resource_name(self):
+        return self.props.get("UserName")
+
+    def fetch_state(self, stack_name, resources):
+        user_name = self.resolve_refs_recursively(stack_name, self.props.get("UserName"), resources)
+        return aws_stack.connect_to_service("iam").get_user(UserName=user_name)["User"]
+
+    def update_resource(self, new_resource, stack_name, resources):
+        props = new_resource["Properties"]
+        client = aws_stack.connect_to_service("iam")
+        return client.update_user(
+            UserName=props.get("UserName"),
+            NewPath=props.get("NewPath") or "",
+            NewUserName=props.get("NewUserName") or "",
+        )
+
```

Comment:
Thanks for this PR @aazbeltran ! I think we still need to add the `get_deploy_templates()` functions, to specify how the resource should be deployed by our CF engine (the function names and parameters below map to boto3 SDK functions).

Can we please add the code below:
```
@staticmethod
def get_deploy_templates():
    return {
        "create": {
            "function": "create_user",
            "parameters": ["Path", "UserName", "PermissionsBoundary", "Tags"],
        },
        "delete": {
            "function": "delete_user",
            "parameters": ["UserName"],
        },
    }
```
(After adding it, you may have to run `make format` to re-format the code and make the linter happy). Thanks!

Reply:
Added 👍🏿